botorch/models/higher_order_gp.py

```diff
@@ -384,6 +385,19 @@ def __init__(
             input_transform.to(train_X)
 
         if outcome_transform is not None:
+            if isinstance(outcome_transform, Standardize) and not isinstance(
+                outcome_transform, FlattenedStandardize
+            ):
+                warnings.warn(
+                    "HigherOrderGP does not support the outcome_transform "
+                    "`Standardize`! Using `FlattenedStandardize` with `output_shape="
+                    f"{train_Y.shape[-2:]} instead.",
+                    RuntimeWarning,
+                )
+                outcome_transform = FlattenedStandardize(
+                    output_shape=train_Y.shape[-2:],
```

Comment:
Just to verify, `FlattenedStandardize` `output_shape` is supposed to be `q x d` dimensions? That's what I understood from existing tests. Adding docstrings to`FlattenedStandardize` may be helpful here (could add in this PR).

Reply:
So it's supposed to be whatever the output dimension of the training data is (which can be multi-dimensional). That is `train_Y` will be `batch_shape x n x output_shape`, where `output_shape` is of dim 1+ and could have an arbitrary number of dimensions (in practice this will be soon hit scalability limits of course) and `batch_shape` is at most one-dimensional (support for multi-level batched models is not implemented yet). To indicate whether a batch dim is present, use `first_dim_is_batch=True`.

This API is probably not ideal - for consistency I would suggest replacing `first_dim_is_batch` by a `num_output_dims` arg that then indicates `len(output_shape)` and thus makes things future-proof for additional batch dimensions should we ever need them. For now we can error out if the passed batch shapes are multi-dimensional. @wjmaddox  any thoughts here? 

Definitely agree on the need for better docstrings though/

@wjmaddox

Reply:
I updated it to infer the `output_shape` from `train_Y`. Also made the shapes of `train_X` and `train_Y` explicit in the docstring.

> for consistency I would suggest replacing first_dim_is_batch by a num_output_dims arg that then indicates len(output_shape)

`first_dim_is_batch` only comes into play in the constructor, so this should be easy to implement. I could make the change and raise an error (`NotImplementedError` ?) if the implied `batch_shape` is multi-dimensional. Does this sound appropriate?

Reply:
Yes that sounds like the right thing to do. Thanks.

Reply:
I think that a `num_output_dims` arg is probably the best bet because it ought to be pretty straightforward to allow for multiple batched output dims if we know how many output dimensions there are.

Reply:
It looks like `num_output_dims` is not necessary at all. Since `train_X` is `batch_shape x n x d` and `train_Y` is `batch_shape x n x output_shape`, we can just infer `num_output_dims = train_Y.dim() - train_X.dim() + 1`. 

Is this correct or is it possible to have `train_X: n x d` with `train_Y: batch_shape x n x output_shape`?

Reply:
I believe that situation (batched Y, non-batched X) occurs when fantasizing, so you do need `num_output_dims`.

Reply:
I didn't think of fantasizing but it looks like `get_fantasy_model` expands the inputs (X) with the batch shape as well. So, this doesn't happen there either. I guess the question is does HOGP support batched Y with non-batched X. The answer to that also seems to be no, since `_validate_tensor_args` (of `GPyTorchModel`) throws an error when I try this.

Reply:
It looks like inferring `num_output_dims = train_Y.dim() - train_X.dim() + 1` is safe. So, I removed `first_dim_is_batch` and replaced it with an inferred `num_output_dims`.